src/llmq/dkgsessionmgr.cpp

```diff
@@ -151,6 +151,8 @@ void CDKGSessionManager::UpdatedBlockTip(const CBlockIndex* pindexNew, bool fIni
     for (auto& qt : dkgSessionHandlers) {
         qt.second.UpdatedBlockTip(pindexNew);
     }
+
+    CleanupOldContributions(pindexNew);
```

Comment:
can this be done on a new thread? and preferably without locking cs_main (or at least not as much as you are)?